src/ILCompiler.Compiler/src/Compiler/RyuJitCompilationBuilder.cs

```diff
@@ -24,13 +24,53 @@ public RyuJitCompilationBuilder(CompilerTypeSystemContext context, CompilationMo
 
         public override CompilationBuilder UseBackendOptions(IEnumerable<string> options)
         {
-            _jitConfig = new JitConfigProvider(options);
+            var builder = new ArrayBuilder<KeyValuePair<string, string>>();
+
+            foreach (string param in options)
+            {
+                int indexOfEquals = param.IndexOf('=');
+
+                // We're skipping bad parameters without reporting.
+                // This is not a mainstream feature that would need to be friendly.
+                // Besides, to really validate this, we would also need to check that the config name is known.
+                if (indexOfEquals < 1)
+                    continue;
+
+                string name = param.Substring(0, indexOfEquals);
+                string value = param.Substring(indexOfEquals + 1);
+
+                builder.Add(new KeyValuePair<string, string>(name, value));
+            }
+
+            _ryujitOptions = builder.ToArray();
+
             return this;
         }
 
         public override ICompilation ToCompilation()
         {
-            return new RyuJitCompilation(CreateDependencyGraph(), _nodeFactory, _compilationRoots, _logger, _jitConfig);
+            ArrayBuilder<CorJitFlag> jitFlagBuilder = new ArrayBuilder<CorJitFlag>();
+
+            switch (_optimizationMode)
+            {
+                case OptimizationMode.None:
+                    jitFlagBuilder.Add(CorJitFlag.CORJIT_FLAG_DEBUG_CODE);
+                    break;
+
+                case OptimizationMode.PreferSize:
+                    jitFlagBuilder.Add(CorJitFlag.CORJIT_FLAG_SIZE_OPT);
+                    break;
+
+                default:
```

Comment:
The RyuJIT default is to not have either of these two flags set. It generates blend of speed and size opts. It may be a good idea to make it the default because it is how it is used everywhere else.